pkg/backend/http/http_client.go

```diff
@@ -80,3 +84,36 @@ var NullHttpClient = func(req *http.Request) (res *http.Response, err error) {
 	defer req.Body.Close()
 	return
 }
+
+func CheckEndpointReachability(ctx context.Context, l log.Entry, endpointURL, license, userAgent, agentID string, timeout time.Duration, transport http.RoundTripper) (timedOut bool, err error) {
+	var request *http.Request
+	if request, err = http.NewRequest("HEAD", endpointURL, nil); err != nil {
+		return false, fmt.Errorf("unable to prepare availability request: %v, error: %s", request, err)
+	}
+
+	request = request.WithContext(ctx)
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Set("User-Agent", userAgent)
+	request.Header.Set(LicenseHeader, license)
+	request.Header.Set(EntityKeyHeader, agentID)
+
+	client := GetHttpClient(timeout, transport)
+
+	// all status codes are acceptable as request has been replied by the endpoint
+	if _, err = client.Do(request); err != nil {
+		if e2, ok := err.(net.Error); ok && (e2.Timeout() || e2.Temporary()) {
+			timedOut = true
+		}
+		if _, ok := err.(*url.Error); ok {
+			// not returning error as this might be a temporal connectivity issue
```

Comment:
Not sure if I get this comment ☝️  as in this case `err` is also being returned 🤷‍♂️

Reply:
leftover from previous work, removed. thanks!